modules/auxiliary/scanner/db2/discovery.rb

```diff
@@ -32,33 +32,37 @@ def run_host(ip)
 
       connect_udp
       udp_sock.put(pkt)
-      res = udp_sock.read(1024).split(/\x00/)
-
-      if (res)
-        report_note(
-          :host   => ip,
-          :proto  => 'udp',
-          :port   => datastore['RPORT'],
-          :type   => 'SERVICE_INFO',
-          :data   => res[2] + "_" + res[1]
-          )
-        report_service(
-          :host => ip,
-          :port => datastore['RPORT'],
-          :proto => 'udp',
-          :name => "ibm-db2",
-          :info => res[2] + "_" + res[1]
-          )
-        print_status("Host #{ip} node name is " + res[2] + " with a product id of " + res[1] )
-      else
+      res = udp_sock.read(1024)
+
+      unless res
         print_error("Unable to determine version info for #{ip}")
+        return
       end
 
-      disconnect_udp
+      res = res.split(/\x00/)
+
+      report_note(
+        :host   => ip,
+        :proto  => 'udp',
+        :port   => datastore['RPORT'],
+        :type   => 'SERVICE_INFO',
+        :data   => res[2] + "_" + res[1]
+        )
+
+      report_service(
+        :host => ip,
+        :port => datastore['RPORT'],
+        :proto => 'udp',
+        :name => "ibm-db2",
+        :info => res[2] + "_" + res[1]
```

Comment:
If there were no `\x00` bytes, both of these indices can be `nil` as well.
```ruby
:info => "#{res[2]}_#{res[1]}"
```
is safer.